main/src/ca/uwaterloo/flix/language/phase/Monomorpher.scala

```diff
@@ -128,21 +128,25 @@ object Monomorpher {
     /**
       * Applies `this` substitution to the given type `tpe`, returning a normalized type.
       *
+      * `keepVariables` can only be `true` if `this.s` is empty. It will avoid removing variables.
```

Comment:
__First__ describe what the flag does. __Then__ describe its invariants.

Reply:
This change seem to erode the subst. abstraction. 

If you want the effects of the Type.Apply cases-- why not extract them into a separate function?

Reply:
__Alternatively__ why not parameterize over the default case?

Reply:
But I much prefer to not break Subst. like this. Better to extract the functionality.